server/src/main/java/org/elasticsearch/ingest/IngestService.java

```diff
@@ -192,120 +192,108 @@ private static Map<String, Processor.Factory> processorFactories(List<IngestPlug
         return Collections.unmodifiableMap(processorFactories);
     }
 
-    public static boolean resolvePipelines(
+    /**
+     * Resolves the potential pipelines (default and final) from the requests or templates associated to the index and then **mutates**
+     * the {@link org.elasticsearch.action.index.IndexRequest} passed object with the pipeline information.
+     *
+     * @param originalRequest Original write request received.
+     * @param indexRequest    The {@link org.elasticsearch.action.index.IndexRequest} object to update.
+     * @param clusterMetadata Cluster metadata from where the pipeline information could be derived.
+     */
+    public static void updateIndexRequestWithPipelines(
```

Comment:
`resolvePipelines` was a good name, and I think it should come back.

The relationship between `resolvePipelines` and `isPipelineResolved` should get second-to-top billing in the docstring, since they're so tightly intertwined with each other.

Tell me about why you're not having it return a boolean anymore? I'm not saying I agree or disagree, but I'd like to understand.

Reply:
my logic is: this is a mutate-this-object-pls™️ method, so, the return value could be confused with a `Hey, I **did/didn't** change your object`. That was my reasoning the first time I read about this method in `TransportBulkAction`.